### DISCUSSION TITLE
EPUBv3 Compliance/Support

### DISCUSSION THREAD

**Discussion by ryanwwest** [koreader/koreader#12528] [Q&A]
I recently learned about [StoryTeller](https://smoores.gitlab.io/storyteller/), "A self-hosted platform for automatically syncing ebooks and audiobooks.". They mention [here](https://smoores.gitlab.io/storyteller/docs/reading-your-books/other-apps) that they are looking for other ereader apps that support EPUB v3 and "Media Overlay specification for EPUB" specifically, but few do. 

I was wondering if KOReader could help them, or if we are not compliant with this? Or any additional relevant information about what level of compliance and support KOReader _does_ have, if any. Thanks.

**Comment by poire-z** [comment]
We may only support EPUB3's ways of handling Toc and reference page numbers, and some of its metadata.
We don't support media overlays or fixed layout stuff, and neither any of the pagination or reading direction stuff.
As for CSS, we support a limited set, and not all that web browsers do, which I think is what EPUB3 expects from renderers.

In my notes about EPUB3, I have these links:
https://wiki.mobileread.com/wiki/EPub_3 other epub3 features
https://github.com/readium/readium-sdk
http://azardi.infogridpacific.com/azardi-feature-map.html
http://apex.infogridpacific.com/df/home.html many posts on epub3 and test suites 
https://www.pagedmedia.org/library/ paged media stuff
https://idpf.github.io/epub3-samples/30/feature-matrix.html

So, no, we're not a good candidate to be tested with (not do we really need to support more of the EPUB3 fancy stuff, it's so rare).

**Comment by pazos** [comment]
EPUB v3 and "Media Overlay specification for EPUB" requires a modern browser, such as chrome, firefox or safari.

>  any additional relevant information about what level of compliance

No compliance at all. KO is still a very good candidate **to be used** to read modern epubs as long as they don't have vertical text and they're reflowable.

Some modern poetry and other interactive docs rely on javascript, which we don't have.
Audio and video content rely on a media engine, which we don't have.

The only apps that reach some type of compliance are web apps running on a browser, directly or by wrapping the engine with the application.
